javagen/src/main/java/com/azure/autorest/util/ClientModelUtil.java

```diff
@@ -63,6 +63,25 @@ public static void getAsyncSyncClients(CodeModel codeModel, ServiceClient servic
                     .packageName(packageName)
                     .serviceClient(serviceClient);
 
+            final List<ConvenienceMethod> convenienceMethods = new ArrayList<>();
+            codeModel.getOperationGroups().stream()
+                    .filter(og -> CoreUtils.isNullOrEmpty(og.getLanguage().getJava().getName()))    // no resource group
+                    .findAny()
+                    .ifPresent(og -> {
+                        og.getOperations().stream()
+                                .filter(o -> generateConvenienceMethods || o.getConvenienceApi() != null)
+                                .forEach(o -> {
+                                    List<ClientMethod> cMethods = Mappers.getClientMethodMapper().map(o, false);
+                                    if (!cMethods.isEmpty()) {
+                                        String methodName = cMethods.iterator().next().getProxyMethod().getName();
+                                        serviceClient.getClientMethods().stream()
+                                                .filter(m -> methodName.equals(m.getProxyMethod().getName()))
+                                                .forEach(m -> convenienceMethods.add(new ConvenienceMethod(m, cMethods)));
+                                    }
+                                });
+                    });
+            builder.convenienceMethods(convenienceMethods);
```

Comment:
this is almost (but slightly different) duplicate code as below on method groups, tolerate it for now.